### ISSUE TITLE
Raise an error when use categorical_crossentropy for binary classification

### ISSUE THREAD

**Issue by haifeng-jin** [keras-team/tf-keras#426]
**System information**.

TensorFlow version (you are using): nightly
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

When use `categorical_crossentropy` for binary classification,
it does not raise any error, but have all the losses equal to 0 during training.
It may confuse the user that the model has some problem, but actually it just because they accidentally used the wrong loss.
See [this colab](https://colab.research.google.com/drive/12UkksHyPajWrB9d7NiFbPWsN47LXWZ0R?usp=sharing) for more details.

So I propose to raise an error if the shape of `y_true` and `y_pred` are all `(None, 1)`.
Unless this use case makes any sense.

**Will this change the current api? How?**
No.

**Who will benefit from this feature?**
ML Beginners or experienced engineer who accidentally used the wrong loss.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing): judge `y_true` and `y_pred` shape and raise error if they are `(None, 1)`.
